src/runtime/Context.cpp

```diff
@@ -101,6 +101,21 @@ void Context::throwException(ExecutionState& state, const Value& exception)
     }
 }
 
+void Context::setGlobalObjectProxy(Object* newGlobalObjectProxy)
+{
+    m_globalObjectProxy = newGlobalObjectProxy;
+
+    // this setter try to update `globalThis` value on GlobalObject
+    SandBox sb(this);
+    sb.run([](ExecutionState& state, void* data) -> Value {
+        Object* newGlobalObjectProxy = (Object*)data;
+        state.context()->globalObject()->defineOwnProperty(state, ObjectPropertyName(state.context()->staticStrings().globalThis),
```

Comment:
Why `globalThis` property is added in a SandBox run?
It seems that we can directly set `globalThis` property without SandBox.

Reply:
Because we don't have special slot for globalThis. the globalThis is plain data property & it can be deleted